### ISSUE TITLE
SAM Predictor.inference/prompt_inference only ever returns a tuple of size 2 when it should be 3 (low resolution logits always missing)

### ISSUE THREAD

**Issue by zamanzamzz** [ultralytics/ultralytics#14062]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Predict

### Bug

I am trying to access the low resolution logits of the SAM predictions from the return value of Predictor.inference as the docs suggest, but it is never there.

Is this expected behaviour?

I would like to use the low resolution logits for the purpose of reusing it for further predictions as a prediction refinement mechanism. However, I came across issue #2661 where it is mentioned that there is no built-in refinement mechanism for SAM. But why else would one use the mask prompt then, which only accepts the low resolution logits from earlier predictions?

### Environment

Ultralytics YOLOv8.1.12 🚀 Python-3.12.3 torch-2.3.0+cu121 CUDA:0 (NVIDIA GeForce RTX 3080 Ti, 12035MiB)
Setup complete ✅ (32 CPUs, 62.7 GB RAM, 660.7/915.5 GB disk)

OS                  Linux-6.1.92-1-MANJARO-x86_64-with-glibc2.39
Environment         Linux
Python              3.12.3
Install             git
RAM                 62.74 GB
CPU                 AMD Ryzen 9 5950X 16-Core Processor
CUDA                12.1

matplotlib          ✅ 3.9.0>=3.3.0
numpy               ✅ 1.26.4>=1.22.2
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.3.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.2>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.3.0>=1.8.0
torchvision         ✅ 0.18.0>=0.9.0
tqdm                ✅ 4.66.2>=4.64.0
psutil              ✅ 5.9.8
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1
pandas              ✅ 2.2.2>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0

### Minimal Reproducible Example

```
from ultralytics.models.sam import Predictor
import cv2

# embed is set to true so that the predictor spits out the result of Predictor.inference
overrides = dict(embed=True, task="segment", mode="predict", imgsz=1024, model="sam_b.pt")
predictor = Predictor(overrides=overrides)

predictor.set_image('test_image.png')

preds = predictor(bboxes=[100, 200, 200, 400])
print(len(preds))
# preds is a tuple of length 2 rather than 3
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
Hi @zamanzamzz,

Thank you for bringing this to our attention and for providing a detailed report. It seems like you're encountering an issue where the `Predictor.inference` method is returning a tuple of size 2 instead of 3, missing the low-resolution logits.

To better assist you, could you please verify the following:
1. Ensure you are using the latest version of the Ultralytics YOLOv8 package. Sometimes, issues are resolved in newer releases.
2. Provide a minimal reproducible example if possible. This helps us replicate the issue on our end and diagnose it more effectively. You can refer to our [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example) guide for more details.

Based on your provided code, it looks like you are correctly setting up the `Predictor` and calling the `inference` method. However, the expected low-resolution logits are not being returned. This might be due to an internal handling within the `prompt_inference` method.

Here's a quick check you can perform to ensure the low-resolution logits are being processed correctly:

```python
from ultralytics.models.sam import Predictor
import cv2

# embed is set to true so that the predictor spits out the result of Predictor.inference
overrides = dict(embed=True, task="segment", mode="predict", imgsz=1024, model="sam_b.pt")
predictor = Predictor(overrides=overrides)

predictor.set_image('test_image.png')

preds = predictor.inference(bboxes=[100, 200, 200, 400])
print(len(preds))  # Check the length of the returned tuple
print(preds)       # Print the entire tuple to inspect its contents
```

If the issue persists, it might be an oversight in the current implementation. In that case, please consider submitting a detailed bug report or a Pull Request (PR) to help us improve the functionality. Your contributions are always welcome!

For further details on the `Predictor` class and its methods, you can refer to the [SAM Predictor Documentation](https://docs.ultralytics.com/reference/models/sam/predict/).

Thank you for your patience and understanding. If you have any more questions or need further assistance, feel free to ask!
